File: shoring/compat/src/main/java/com/panopset/compat/Jsonop.java
package com.panopset.compat;

import java.lang.reflect.Type;
import com.google.gson.Gson;

public class Jsonop {

  public String toJson(Object obj) {
    return new Gson().toJson(obj);
  }

  public Object fromJson(String json, Class<?> clazz) {
    return new Gson().fromJson(json, clazz);
  }

  /**
   *  https://stackoverflow.com/questions/5554217.
   */
  public Object listFromJson(String json, Type type) {
    return new Gson().fromJson(json, type);
  }
}

File: code/shoring/compat/src/main/kotlin/com/panopset/compat/Jsonop.kt
package com.panopset.compat

import com.google.gson.Gson
import java.lang.reflect.Type

class Jsonop<T> {
    fun toJson(obj: T): String {
        return Gson().toJson(obj)
    }

    /**
     * https://stackoverflow.com/questions/5554217.
     */
    fun fromJson(json: String, type: Type): T {
        return Gson().fromJson(json, type)
    }
}
